docs/CONTEXT.md

```diff
@@ -0,0 +1,96 @@
+# Context object
+`HullApp` applies multiple middleware to the request handler.
+The result is `req.hull` object which is the Context Object - a set of modules to work in the context of current organization and connector instance.
+
+```javascript
+{
+  client: {
+    logger: {},
+  },
+  ship: {},
+  token: "",
+  segments: [],
+  hostname: req.hostname,
+  cache: {},
+  enqueue: () => {},
+  metric: {},
+  service: {},
+  message: {},
+  notification: {}
+}
+```
+
+#### client
+[Hull API client](../README.md) initialized to work with current organization
+
+#### ship
+Ship object with manifest information and `private_settings`
+
+#### token
+A string with encoded credentials
+
+#### segments
+An array of segments information
+
+#### hostname
+Hostname of the current request.
+
+#### cache
+[Caching object](../src/infra/cache/ship-cache.js)
+
+#### enqueue
+A [function](../src/infra/queue/enqueue.js) to perform tasks in async manner.
+
+#### metric
+An [object](../src/infra/instrumentation/metric-agent.js) to send data to metrics service.
+
+#### service
+A namespace reserved for connector developer to inject a custom logic, which needs to work in the context.
```

Comment:
need much more explanation, and examples on what this does and why it exists. Have no idea what this is and what it's for.